srml/staking/src/lib.rs

```diff
@@ -1564,3 +1470,80 @@ impl<T: Trait> SelectInitialValidators<T::AccountId> for Module<T> {
 		<Module<T>>::select_validators().1
 	}
 }
+
+impl <T: Trait> OnOffenceHandler<T::AccountId, session::historical::IdentificationTuple<T>> for Module<T> where
+	T: session::Trait<ValidatorId = <T as system::Trait>::AccountId>,
+	T: session::historical::Trait<
+		FullIdentification = Exposure<<T as system::Trait>::AccountId, BalanceOf<T>>,
+		FullIdentificationOf = ExposureOf<T>,
+	>,
+	T::SessionHandler: session::SessionHandler<<T as system::Trait>::AccountId>,
+	T::OnSessionEnding: session::OnSessionEnding<<T as system::Trait>::AccountId>,
+	T::SelectInitialValidators: session::SelectInitialValidators<<T as system::Trait>::AccountId>,
+	T::ValidatorIdOf: Convert<<T as system::Trait>::AccountId, Option<<T as system::Trait>::AccountId>>
+{
+	fn on_offence(
+		offenders: &[OffenceDetails<T::AccountId, session::historical::IdentificationTuple<T>>],
+		slash_fraction: &[Perbill],
+	) {
+		let mut remaining_imbalance = <NegativeImbalanceOf<T>>::zero();
+		let slash_reward_fraction = SlashRewardFraction::get();
+
+		for (details, slash_fraction) in offenders.iter().zip(slash_fraction) {
+			let stash = &details.offender.0;
+			let exposure = &details.offender.1;
+
+			// Early exit if validator is invulnerable.
+			if Self::invulnerables().contains(stash) {
+				return
```

Comment:
```suggestion
				continue
```

One offender being invulnerable shouldn't get the others off

Reply:
Good point

Reply:
Fixed and add a test in https://github.com/paritytech/substrate/pull/3322/commits/53b2c6449b41ced1dd3cfc5330e59e004187af78